aws/src/main/java/org/apache/iceberg/aws/s3/S3OutputStream.java

```diff
@@ -87,17 +190,105 @@ public void close() throws IOException {
 
     super.close();
     closed = true;
+    currentStagingFile = null;
 
     try {
       stream.close();
 
+      completeUploads();
+    } finally {
+      stagingFiles.forEach(f -> {
+        if (f.exists() && !f.delete()) {
+          LOG.warn("Could not delete temporary file: {}", f);
+        }
+      });
+    }
+  }
+
+  private void initializeMultiPartUpload() {
+    multipartUploadId = s3.createMultipartUpload(CreateMultipartUploadRequest.builder()
```

Comment:
s3a supports canned acls for s3 requests, not sure if this should also support them?

My team had to use them once for writing from within account A to a bucket owned by bucket B

We also had to add it to rdblue/s3committer:
https://github.com/rdblue/s3committer/blob/master/src/main/java/com/netflix/bdp/s3/S3Util.java#L73
```
    InitiateMultipartUploadRequest initiateMultipartUploadRequest = new InitiateMultipartUploadRequest(bucket, key)
          .withCannedACL(CannedAccessControlList.BucketOwnerFullControl);
    InitiateMultipartUploadResult initiate = client.initiateMultipartUpload(initiateMultipartUploadRequest);
```

Reply:
Interesting, I didn't even know about canned acls.  Looks like the builder does support it (via `.acl()`), so that's probably  another thing we can plumb through.

Reply:
For sure, it should probably go in another PR.

Should I file an issue? Or is the issue area mainly for items people are currently working on a PR for? I probably won't work on that any time soon.

(We only set canned acls to backport iceberg tables from one account to hive tables in a legacy account, so my team doesn't need this in order to move onto S3FileIO)

Reply:
Yes I think canned ACL is a feature for another PR, and I was planning to do the PR after #1754 , let's wait until that is checked in so that we have a stable interface for adding new things.

Reply:
I'd say wait until you feel like you need it and then file an issue.  Filing the issue doesn't mean you have to work on it, but this feels like one of the many features of the client that may not be used very much.  It should be trivial to add, so if you find that you do need it, we can easily add it then.

Reply:
Yes. This is one of the things I planned to port, I am fine either me or @johnclara do it.

Reply:
@jackye1995 if you were already planning on it, go ahead! :) my team doesn't need it any time soon but it would be good for people who run into a similar issue